include/jemalloc/internal/cache_bin.h

```diff
@@ -182,6 +182,36 @@ cache_bin_array_descriptor_init(cache_bin_array_descriptor_t *descriptor,
 	descriptor->bins_large = bins_large;
 }
 
+JEMALLOC_ALWAYS_INLINE void *
+cache_bin_alloc_opt(cache_bin_t *bin, bool *success) {
```

Comment:
Let's do some refactoring to reduce duplicated code:
- add a `cache_bin_alloc_easy_impl(bin, success, const bool handle_low_water)`;
- and the prev alloc_easy will have the last arg as true, the new one will have false;
- then within the unlikely branch, we can decide if low_water related stuff should be considered.

The naming is hard. Not a fan of opt since it also means option in our code. Maybe call the new one `cache_bin_alloc_oversimplified`?